src/core/qgsvectorfilewriter.cpp

```diff
@@ -674,1211 +674,1243 @@ OGRGeometryH QgsVectorFileWriter::createEmptyGeometry( QgsWkbTypes::Type wkbType
+                               layerOptions,
+                               QStringLiteral( "UTF-8" )
+                             )
+                           );
+
+      // GeoPackage
+      datasetOptions.clear();
+      layerOptions.clear();
+
+      layerOptions.insert( QStringLiteral( "IDENTIFIER" ), new QgsVectorFileWriter::StringOption(
+                             QObject::tr( "Human-readable identifier (e.g. short name) for the layer content" ),
+                             QLatin1String( "" )  // Default value
+                           ) );
+
+      layerOptions.insert( QStringLiteral( "DESCRIPTION" ), new QgsVectorFileWriter::StringOption(
+                             QObject::tr( "Human-readable description for the layer content" ),
+                             QLatin1String( "" )  // Default value
+                           ) );
+
+      layerOptions.insert( QStringLiteral( "FID" ), new QgsVectorFileWriter::StringOption(
+                             QObject::tr( "Name for the feature identifier column" ),
+                             QStringLiteral( "fid" )  // Default value
+                           ) );
+
+      layerOptions.insert( QStringLiteral( "GEOMETRY_NAME" ), new QgsVectorFileWriter::StringOption(
+                             QObject::tr( "Name for the geometry column" ),
+                             QStringLiteral( "geom" )  // Default value
+                           ) );
+
+      layerOptions.insert( QStringLiteral( "SPATIAL_INDEX" ), new QgsVectorFileWriter::BoolOption(
+                             QObject::tr( "If a spatial index must be created." ),
+                             true  // Default value
+                           ) );
+
+      driverMetadata.insert( QStringLiteral( "GPKG" ),
+                             QgsVectorFileWriter::MetaData(
+                               QStringLiteral( "GeoPackage" ),
+                               QObject::tr( "GeoPackage" ),
+                               QStringLiteral( "*.gpkg" ),
+                               QStringLiteral( "gpkg" ),
+                               datasetOptions,
+                               layerOptions,
+                               QStringLiteral( "UTF-8" )
+                             )
+                           );
+
+      // Generic Mapping Tools [GMT]
+      datasetOptions.clear();
+      layerOptions.clear();
+
+      driverMetadata.insert( QStringLiteral( "GMT" ),
+                             QgsVectorFileWriter::MetaData(
+                               QStringLiteral( "Generic Mapping Tools [GMT]" ),
+                               QObject::tr( "Generic Mapping Tools [GMT]" ),
+                               QStringLiteral( "*.gmt" ),
+                               QStringLiteral( "gmt" ),
+                               datasetOptions,
+                               layerOptions
+                             )
+                           );
+
+      // GPS eXchange Format [GPX]
+  
```

Comment:
Can't we just put all the init code into the constructor and skip this conditional initialisation?

Reply:
Yep, good approach. This was a iterative solution :)

Reply:
Fixed